### PR TITLE
Collect reporter artifacts in a JustAfterEach

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

In order to make it finally possible to clean up environments as
expected in a `AfterEach` the reporters are now collecting the
environment info in a `JustAfterEach`. This means that it does no
longer matter if `BeforeTestCleanup` is called in a `BeforeEach` or
`AfterEach`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/tests_suite_test.go [hunk 1]
```diff
@@ -86,3 +93,11 @@ func getMaxFailsFromEnv() int {
 
 	return maxFails
 }
+
+// Collect info directly after each `It` execution with our reporters
+// to collect the state directly after the spec.
+var _ = JustAfterEach(func() {
```
Comment (vasiliy-ul) [inline tests/tests_suite_test.go:99]: Probably there is some confusion in the PR description and commit message:
> Collect reporter artifacts in a Just**Before**Each

but maybe should be

> Collect reporter artifacts in a Just**After**Each

**Review by vasiliy-ul — COMMENTED** [review]
Looks great. Just a small comment regarding the PR/commit descritpion.

File: tests/tests_suite_test.go (hunk 1 above)
Comment (rmohr) [reply tests/tests_suite_test.go:99]: Great catch. Fixed in the commit and the PR description.

**Pull request merged**